src/lua/zencode_ethereum.lua

```diff
@@ -388,3 +388,21 @@ IfWhen("verify the '' has a ethereum signature in '' by ''", function(doc, sig,
        'The ethereum signature by '..by..' is not authentic'
     )
 end)
+
+When("use the ethereum transaction to run '' using ''", function(m, p)
+
+    local method = have(m)
+    local params = have(p)
+    ZEN.assert(
+        method.name and method.input and type(method.input) == "table",
+       'method must contain the keys method and input'
+    )
+    local input = deepmap(function(o)
```

Comment:
here is clear that method.input is supposed to be an array (or dictionary?) of strings, so we should also check the CODEC records for that

Reply:
`input` is a key inside the method dictionary, so it doesn't have a CODEC.
Should we introduce a schema for a method? This is not something standard, it is a dictionary introduced by us
```
{
   name: string,
   input: [string],
   output: [string]
}
```